addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusProvider.java

```diff
@@ -0,0 +1,723 @@
+     * {@link getTivoStatus} checks if the current connection is "Keep Open" and connected. Connect, query and close as
+     * required.
+     *
+     * @return {@link TivoStatusData} object.
+     */
+
+    private TivoStatusData statusGet() {
+
+        boolean oC = false;
+        TivoStatusData rtnVal = null;
+        int retryCount = 2;
+
+        // Will open a connection prior to operation if we are not keeping one open.
+        if (!connTestSocket() || !tivoConfigData.isCfgKeepConnOpen()) {
+            oC = connTivoConnectRetry(true);
+        }
+
+        if (tivoStatusData.getConnOK() > 1) {
+            rtnVal = statusGetRetry(retryCount);
+        }
+
+        // Will close the connection after operation if we are not keeping one open.
+        if (!tivoConfigData.isCfgKeepConnOpen() && oC && !tivoStatusData.getChScan()) {
+            connTivoConnectRetry(false);
+        }
+        return rtnVal;
+    }
+
+    /**
+     * {@link statusGetRetry} queries the TiVo device for the LAST status message and parses the resultant data.
+     * <p>
+     *
+     * When a command is executed, the system either returns a new CH_STATUS message or any error. Most commands do not
+     * return successful status codes (with the exception of a channel change).
+     * <p>
+     *
+     * @param retryCount integer, number of times a message retrieval will be attempted if a communication error occurs
+     * @return {@link TivoStatusData} object containing the result of the connection to the DVR.
+     */
+    private TivoStatusData statusGetRetry(int retryCount) {
+
+        // Queries the TiVo device for the current status. Status of previous command is displayed for 3-4 seconds.
+        // COMMAND_TIMEOUT is then displayed (equivalent to 'waiting for next command').
+
+        logger.debug(" statusGetRetry '{}' - reading from TiVo (retry @ '{}')", tivoConfigData.getCfgIdentifier(),
+                retryCount);
+
+        String line = null;
+        String tivoStatus = null;
+
+        // If were not connected we can't provide a status
+        if (!connTestSocket()) {
+            logger.error(" TiVo '{}' - called statusGetRetry but not connected!", tivoConfigData.getCfgIdentifier());
+            return new TivoStatusData(false, -1, "NOT_CONNECTED", false, 0);
+        }
+
+        try {
+            while ((line = tivoIOReadStatus.readLine()) != null) {
+                logger.debug(" statusGetRetry read: '{}'", line);
+                if (!line.isEmpty() && !line.trim().equalsIgnoreCase("COMMAND_TIMEOUT")) {
+                    // use this line
+                    tivoStatus = line;
+                } else {
+                    logger.debug(" statusGetRetry ignored line: '{}'", line);
+                }
+
+            }
+        } catch (IOException e) {
+            logger.debug(" statusGetRetry '{}' - I/O exception: '{}'", tivoConfigData.getCfgIdentifier(), e);
+            if (e.getMessage().equalsIgnoreCase("READ TIMED OUT")) {
```

Comment:
Please check the exception type instead i.e. have different catches this is not the way to do this.